[Bugfix] Update UI and handle new navigation

TackAdam opened the pull request:
### Description
1. Updates the UI with the compressed elements and formatting.
2. Add condition for new navigation to prevent double titles.
3. Update testing to handle chrome prop for new navigation check.

New navigation 
| Before | After |
| ------ | ----- |
| ![Before Image 1](https://github.com/user-attachments/assets/76e2edbc-c79d-4f27-aeda-cb3099cb0508) | ![After Image 1](https://github.com/user-attachments/assets/a594fa55-2893-4f11-a030-86e3428066d0) |
| ![Before Image 2](https://github.com/user-attachments/assets/352d621d-7760-4403-990c-e210f618eefc) | ![After Image 2](https://github.com/user-attachments/assets/6d5249ff-91c8-44b4-8cbe-a58351163051) |
| ![Before Image 3](https://github.com/user-attachments/assets/924b00c4-3b7b-46be-9077-1a36da4b4e8a) | ![After Image 3](https://github.com/user-attachments/assets/8f881032-7c65-44d0-bcf3-0170b6d36aee) |
| ![Before Image 4](https://github.com/user-attachments/assets/6a2a6d48-94e4-47e8-8c76-ab9c44b5be85) | ![After Image 4](https://github.com/user-attachments/assets/6f614ef0-8408-45e1-99c4-0b0dc71d9fd1) |
| ![Before Image 5](https://github.com/user-attachments/assets/6410f018-8181-4a10-923a-54dd206ae415) | ![After Image 5](https://github.com/user-attachments/assets/871afb6a-10ac-4be6-8f52-190ad55b44c7) |

Old Navigation
| Before | After |
| ------ | ----- |
| ![Before Image 1](https://github.com/user-attachments/assets/862d2601-8a8f-4317-83a2-f3bcd21d7ddb) | ![After Image 1](https://github.com/user-attachments/assets/115b4764-0407-4237-94c1-0f1f53f6c1dd) |
| ![Before Image 2](https://github.com/user-attachments/assets/d1232bfd-7f16-4786-bd61-33479ee82fab) | ![After Image 2](https://github.com/user-attachments/assets/a7e41464-2776-49b3-a036-f9b49171728f) |
| ![Before Image 3](https://github.com/user-attachments/assets/ef182714-57b8-44c5-8254-0897e967da98) | ![After Image 3](https://github.com/user-attachments/assets/f349d76b-60e2-4211-87ab-df760571a31f) |
| ![Before Image 4](https://github.com/user-attachments/assets/508896ca-aeac-47bf-8be2-1ff48cfc1168) | ![After Image 4](https://github.com/user-attachments/assets/42c2a0d8-ebd6-4b4f-87c3-9657d2c0dd3e) |
| ![Before Image 5](https://github.com/user-attachments/assets/56b95c5a-7b79-47dd-a7cc-c3f3b9f8a652) | ![After Image 5](https://github.com/user-attachments/assets/28124696-7de3-4413-8f06-246a07cf618d) |

### Issues Resolved

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

derek-ho commented on public/components/report_definitions/create/create_report_definition.tsx:
```diff
@@ -301,12 +304,12 @@ export function CreateReport(props: { [x: string]: any; setBreadcrumbs?: any; ht
           </h1>
         </EuiTitle>
-        <EuiSpacer />
+        <EuiSpacer size='s'/>
```
QQ: Do we want to remove the extra spacing if the h1 is not present/empty? Let me know if this makes sense

TackAdam replied:
Good catch, fixed with the following 
{!getNavGroupEnabled && <EuiSpacer size='s' />}
Very minor look change for navigation enabled but makes sense!

derek-ho commented on public/components/main/report_definition_details/report_definition_details.tsx:
```diff
@@ -669,169 +668,167 @@ export function ReportDefinitionDetails(props: { match?: any; setBreadcrumbs?: a
+          )}
+        </h1>
+      </EuiTitle>
+      <EuiSpacer size="m" />
```
Same question as below

TackAdam replied:
Fixed with {!getNavGroupEnabled && <EuiSpacer size='s' />}

derek-ho reviewed (approved):
Mostly LGTM

ps48 commented:
Just one callout under the new nav, the table headers look as big as the title. May be we can reduce their size under the new nav?

<img width="396" alt="Screenshot 2024-09-04 at 1 41 21 PM" src="https://github.com/user-attachments/assets/58e75390-cecd-43bb-8bb7-3158a19c0323">

ps48 reviewed (approved):
Nothing blocking thanks for taking this up

TackAdam commented:
Adjusted spacing in old and new navigation. Added tittle to the old nav for it to match
New nav:
<img width="1210" alt="NewNav" src="https://github.com/user-attachments/assets/719ac485-a207-4a76-a36f-a265682b0acd">

Old nav:
<img width="1217" alt="OldNav" src="https://github.com/user-attachments/assets/8581925f-0903-4875-a802-c6b12bc2aed3">
